### PR TITLE
chroot: add `disable_root_check` option

### PR DESCRIPTION
**Pull request by new-sashok724**
##### SUMMARY
Some chroot executables, like `arch-chroot`, support unshare mode (-N), which does not require root. This PR adds an option to disable the root check in the chroot connection plugin.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
`plugins/connection/chroot.py`

### CODE REVIEW

File: plugins/connection/chroot.py [hunk 1]
```diff
@@ -46,6 +46,17 @@
...
+        description:
+            - Do not check that the user is not root.
+        ini:
+          - section: chroot_connection
+            key: disable_root_check
+        env:
+          - name: ANSIBLE_CHROOT_DISABLE_ROOT_CHECK
+        vars:
+          - name: ansible_chroot_disable_root_check
+        default: false
```
Comment (felixfontein) [inline plugins/connection/chroot.py:59]:
```suggestion
        default: false
        type: bool
        version_added: 7.3.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment). Thanks.

**Comment by new-sashok724** [comment]
Added.

File: changelogs/fragments/7099-chroot-disable-root-check-option.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "chroot - add ``disable_root_check`` option (https://github.com/ansible-collections/community.general/pull/7099)."
```
Comment (felixfontein) [inline changelogs/fragments/7099-chroot-disable-root-check-option.yml:2]:
```suggestion
  - "chroot connection plugin - add ``disable_root_check`` option (https://github.com/ansible-collections/community.general/pull/7099)."
```

**Comment by felixfontein** [comment]
The remaining CI failures are unrelated.

Looks good to me, if nobody objects I'll merge this on Monday.

**Comment by felixfontein** [comment]
@new-sashok724 thanks for your contribution!

**Pull request merged**